### ISSUE TITLE
RMSE or MSE?

### ISSUE THREAD

**Issue by kamil-oster** [KindXiaoming/pykan#305]
Hi!
I've been looking through the source code and found that the default loss function is a bit ambiguous.
In training, the optimiser users MSE in the closure:

`loss_fn = loss_fn_eval = lambda x, y: torch.mean((x - y) ** 2)`

The comment section clearly says that it's RMSE which makes sense as later on pbar take torch.sqrt (as well as having that in results dict):

`pbar.set_description("train loss: %.2e | test loss: %.2e | reg: %.2e " % (torch.sqrt(train_loss).cpu().detach().numpy(), torch.sqrt(test_loss).cpu().detach().numpy(), reg_.cpu().detach().numpy()))`

```
results['train_loss'].append(torch.sqrt(train_loss).cpu().detach().numpy())
results['test_loss'].append(torch.sqrt(test_loss).cpu().detach().numpy())
```

 I wonder if there is a reason why one loss function is not used across train function? Or have I missed something?

Thanks, Kamil

**Comment by KindXiaoming** [comment]
Hi, just because RMSE has the same unit (if we think of physical variables, they have units) as the target variables, while MSE  has an extra square. So arguably RMSE is more "physical" than MSE. But this is just a personal choice, not necessary I guess. :)
